modules/core/04-channel/keeper/upgrade.go

```diff
@@ -275,16 +267,14 @@ func (k Keeper) WriteUpgradeTryChannel(
 	emitChannelUpgradeTryEvent(ctx, portID, channelID, upgradeSequence, channelUpgrade)
 }
 
-// TODO: should we pull out the error receipt logic from this function? They seem like two discrete operations.
-
-// RestoreChannelAndWriteErrorReceipt restores the given channel to the state prior to upgrade.
-func (k Keeper) RestoreChannelAndWriteErrorReceipt(ctx sdk.Context, portID, channelID string, upgradeSequence uint64, err error) error {
-	errorReceipt := types.NewErrorReceipt(upgradeSequence, err)
-	k.SetUpgradeErrorReceipt(ctx, portID, channelID, errorReceipt)
-
+// RestoreChannel restores the given channel to the state prior to upgrade.
+func (k Keeper) RestoreChannel(ctx sdk.Context, portID, channelID string) error {
```

Comment:
If this function is now called only at the message server layer maybe we can either:
- Move the call to `OnChanUpgradeRestore` also to the message server layer;
- or pass `cbs` to this function and call the callback here (and thus not needing to retrieve the callback here), since `cbs` should already be known at the message server layer, right?